src/ngraph/validation_util.cpp

```diff
@@ -628,3 +628,174 @@ void ngraph::infer_auto_padding(const Shape& image_shape,
         padding_above.push_back(pad_type == op::PadType::SAME_UPPER ? padding_rhs : padding_lhs);
     }
 }
+
+PartialShape ngraph::infer_slice_shape(const Node* node,
```

Comment:
attn @jbobba: I moved this from DynSlice into validation_util (so that DynReplaceSlice can use it), and extended it to support partiality.